src/Intl/Grapheme/Grapheme.php

```diff
@@ -205,13 +205,24 @@ private static function grapheme_position($s, $needle, $offset, $mode)
             }
         }
 
-        switch ($mode) {
-            case 0: $needle = iconv_strpos($s, $needle, 0, 'UTF-8'); break;
-            case 1: $needle = mb_stripos($s, $needle, 0, 'UTF-8'); break;
-            case 2: $needle = iconv_strrpos($s, $needle, 'UTF-8'); break;
-            default: $needle = mb_strripos($s, $needle, 0, 'UTF-8'); break;
+        // As UTF-8 is self-synchronizing, and we have ensured the strings are valid UTF-8,
+        // we can use normal binary string functions here. For case-insensitive searches,
+        // case fold the strings first.
+        $caseInsensitive = $mode & 1;
+        $reverse = $mode & 2;
+        if ($caseInsensitive) {
+            // Use the same case folding mode as mbstring does for mb_stripos().
+            // We could also use MB_CASE_FOLD (without _SIMPLE) for better results.
```

Comment:
why don't we do it if it yields better results?

Reply:
I'm not sure that would actually be okay in this context ... For example, let's say we have a string `XßYß` and search for `SS` from the end. Then non-simple case-folding would convert the string into something like `XSSYSS` and we'd find it. But at that point the grapheme offset to the second SS would be 4, rather than the 3 you would expect based on the original string.

So it seems safer to stick with SIMPLE folding, which does not change the length of the string.

Reply:
makes sense, thanks!